spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -547,12 +548,21 @@ struct ProductionVisitor
         return stop;
     }
 
-    void preParseField(const Production& /* i */, const production::Meta& meta) {
+    std::optional<Expression> preParseField(const Production& /* i */, const production::Meta& meta) {
         const auto& field = meta.field();
         assert(field); // Must only be called if we have a field.
 
         HILTI_DEBUG(spicy::logging::debug::ParserBuilder, fmt("- pre-parse field: %s", field->id()));
 
+        // If the field holds a container we expect to see the offset of the field, not the individual container
+        // elements inside e.g., this unit's fields hooks. Store the value before parsing of a container starts so we
+        // can restore it later.
+        std::optional<Expression> pre_container_offset;
+        if ( field && type::isIterable(field->parseType()) && ! meta.container() ) {
```

Comment:
Not completely sure, but I think it should work to use this:

```suggestion
        if ( field->isContainer() ) {
```

Reply:
This works well for the test I added, but behaves differently in that the offsets seen in the test `spicy.types.unit.trimming` does not change anymore. This seems better.

Reply:
I'll check for `field && field->isContainer()` here.